### DISCUSSION TITLE
Allow searching in a specific commit in Github CodeSearch (cs.github.com)

### DISCUSSION THREAD

**Discussion by poliorcetics** [community/community#27577] [Other Feature Feedback, Questions, & Ideas]
I wanted to search for code in a specific release of `windows-rs` but could not find an option to search within a specific commit in Github CodeSearch (https://cs.github.com).

I saw other discussions in relation with commits but I don't think they asked for that exactly

**Comment by look** [comment]
Can you clarify what you mean by this? Do you mean search the diff of that commit, or search the repo using the tree as it exists in that commit?

**Reply by poliorcetics** [reply]
The second, "search the repo using the tree as it exists in that commit"

**Reply by look** [reply]
Thanks for clarifying. This is on our radar, but I can't share specifics about when it will be ready. We are focusing now on scaling the system to full GitHub scale.

**Comment by airtower-luna** [comment]
If you have (or clone) a local copy of the repository you can do that with [`git grep`](https://git-scm.com/docs/git-grep), like so:

```sh
git grep pico 0b66a35535742d14f7af2e1d4e22914ed9879eb9
```

In that case `pico` will be searched in all files of the commit `0b66a35535742d14f7af2e1d4e22914ed9879eb9`.

**Reply by poliorcetics** [reply]
Yes, but the entire point of `cs.github.com` is to not have to clone the repo. Having it online also makes it easier to share links to code

**Reply by airtower-luna** [reply]
Then I recommend you mention such context for questions in the future, "how can I search in a commit" is a very different question from "how can I search in a commit inside the `cs.github.com` interface".

**Reply by poliorcetics** [reply]
Isn't that what the "[Code Search and Navigation](https://github.com/orgs/community/discussions/categories/code-search-and-navigation)" category is for ? I was sent here when I clicked on "give feedback" using Github CS. If the feedback is out amongst more general questions there should be a warning else it's really confusing.

**Reply by look** [reply]
@poliorcetics is correct, this category is for feedback about cs.github.com and the related GitHub code navigation feature. The naming is confusing for a lot of people though.

**Reply by airtower-luna** [reply]
Thanks for the clarification!
